Improve Android privacy policy

tomasz1986 opened the pull request:
Change the format and wording to make it a little more formal, so that
it becomes easier to understand by the people responsible for the Google
Play review process.

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

imsodin reviewed (commented):
Thanks for looking into this. Can you please rebase to current main branch and add line breaks after sentences: I added them to be able to sensibly review the text.

I am not a big fan of the subtitles. In my opinion paragraphs are enough to separate this little text. And I don't want to draw too much attention to the "Syncthing" part - I don't want google to notice that and decide they want the linked info directly in this document.  
Definitely the "Permissions" title needs to go: Google's wording is about location data, "permission" isn't ever used. The two issues (location permission and data privacy) are intrinsically linked for us, as for both it's a google play restriction for something we don't deal with. However they are not the same in general/from google's perspective. I wondered whether I should remove the explanation about why we need location permission despite not using location data entirely, but considered it's still valuable information (to answer "why are we even talking about location data if we don't collect any").

tomasz1986 commented:
> I don't want google to notice that and decide they want the linked info directly in this document.

I have a feeling that they will notice and find it problematic later anyway…

> I am not a big fan of the subtitles. In my opinion paragraphs are enough to separate this little text.

The text is short, but I had just checked the privacy policies of some other bigger software, e.g. Brave or Firefox (FOSS) and Dropbox (proprietary), and all of them use this style of big headings with short explanations. I figured that this was probably what Google liked and would be more likely to accept :sweat_smile:.

Anyhow, I have removed the headings and modified some of the text, i.e. mainly shortened the Syncthing part a little bit.

tomasz1986 committed:
Improve Android privacy policy

Change the format and wording to make it a little more formal, so that
it becomes easier to understand by the people responsible for the Google
Play review process.

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

imsodin commented on content/android-privacy-policy.md:
```diff
@@ -1,14 +1,20 @@
-It requires the permission to access the location in the background, to look up the currently active Wi-Fi's SSID.
-That information is used to enable syncing only on user-configurable Wi-Fi's.
-The app never actually looks up your location in the process, and thus doesn't use or store location data.
+The location permission is used only to access background location in order to check for an SSID of the currently active WiFi.
```
What's "background location". And we don't access location, so why say we do. Also the permissions "location in the background" is an entity, a phrase used by Google - having that verbatim is more important than nice sentences :)

tomasz1986 replied:
I took the phrase directly from Google… (see https://android-developers.googleblog.com/2020/02/safer-location-access.html, "Preventing unnecessary access to background location").

The accessing part I can change, but WiFi is in fact a part of location in the eyes of Google, and Syncthing does access it, so…

imsodin commented on content/android-privacy-policy.md:
```diff
@@ -1,14 +1,20 @@
-That information is used to enable syncing only on user-configurable Wi-Fi's.
-The app never actually looks up your location in the process, and thus doesn't use or store location data.
+The location permission is used only to access background location in order to check for an SSID of the currently active WiFi.
+The sole purpose of this information is to limit synchronisation to user-configured WiFi connections.
```
The text in the app uses "Wi-Fi", so I stuck with that on purpose.

imsodin commented on content/android-privacy-policy.md:
```diff
@@ -1,14 +1,20 @@
 
-The app uses the camera solely to scan QR-codes to enter a device ID.
-Pictures are not stored in the process.
+The app does not store any camera data.
```
camera data?

tomasz1986 replied:
Yeah, I just copied the sentence from the location part. It is not really necessary.

imsodin commented on content/android-privacy-policy.md:
```diff
@@ -1,14 +1,20 @@
-It does run Syncthing, which may collect anonymous usage-data only if you actively opt-in to it or run betas/release candidates.
-This documentation article has specific information on this: https://docs.syncthing.net/users/security.html
+Syncthing-Android runs Syncthing, which may collect anonymous usage data related to the usage of the application.
+In order for this to happen, the user is required to either consciously opt in or use beta/release builds.
```
```suggestion
In order for this to happen, the user is required to either consciously opt in or use beta/candidate releases.
```

imsodin reviewed (changes requested):
To be frank: While this improves a few ugly/weird sentences of mine, it also introduces incorrect changes and new weirdness. I don't see it substantially improving the privacy policy and I don't care about the privacy policy beyond what Google thinks about it - so I am not interested in a lengthy review/optimisation process. I only thought about this and wrote it down after doing the individual comments below - if you feel like parts or all of this are important/necessary to pass googles review, please explain why and I'll consider continuing the review process. Otherwise I am more inclined to close this PR, i.e. there's no point in spending time to address comments below.

tomasz1986 commented:
> To be frank: While this improves a few ugly/weird sentences of mine, it also introduces incorrect changes and new weirdness. I don't see it substantially improving the privacy policy and I don't care about the privacy policy beyond what Google thinks about it - so I am not interested in a lengthy review/optimisation process. I only thought about this and wrote it down after doing the individual comments below - if you feel like parts or all of this are important/necessary to pass googles review, please explain why and I'll consider continuing the review process. Otherwise I am more inclined to close this PR, i.e. there's no point in spending time to address comments below.

Seeing the comment, I am not really sure what I ought to do then. Should I still fix those few sentences? I do not want to spend time on something that will eventually go to the trash can either…

I just have a feeling that the whole privacy policy needs to be more thorough and more detailed in order to pass. Otherwise this will end up going back and forth with no ending on the horizon with each new Syncthing release.

For the record, these are samples of some privacy policies that Google definitely accepts. All are super long and extremely detailed. It almost seems as if they write them like that on purpose, so that nobody will have patience to read the whole thing.

https://brave.com/privacy/browser
https://mozilla.org/en-US/privacy/firefox
https://dropbox.com/privacy

**Edit**: I have fixed the text and removed the unnecessary parts nevertheless. It should be much more similar to the original text right now. I have just tried to make the sentences more natural.

tomasz1986 committed:
wip

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

tomasz1986 committed:
wip

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

tomasz1986 committed:
wip

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

tomasz1986 committed:
wip

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

tomasz1986 committed:
wip

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

tomasz1986 committed:
wip

Signed-off-by: Tomasz Wilczyński <twilczynski@naver.com>

imsodin reviewed (approved):
I basically just didn't want to get into any more review with this unless I see strong reasons to assume it's going to change the Google Play outcome. Luckily with the latest changes I agree with the changes, thus no more review is necessary :)

Thanks for helping out and sorry for being a it grumpy - this entire process (as in google play policy shenanigans) isn't what I like about working on Syncthing (to put it mildly).

tomasz1986 commented:
No problem 🙂.

I often tend to make more changes than necessary :grimacing:. Let us hope that the review process will be successful this time.

But really, I would not be surprised if the three privacy policies that I mentioned above were written by actual lawyers. Especially the Dropbox policy, that one was definitely written/reviewed by a lawyer. I guess that this only shows the complexity and how much of a mine field the Google Play has become.
